common/queue/src/main/java/org/thingsboard/server/queue/discovery/ZkDiscoveryService.java

```diff
@@ -290,8 +297,30 @@ public void childEvent(CuratorFramework curatorFramework, PathChildrenCacheEvent
         log.debug("Processing [{}] event for [{}]", pathChildrenCacheEvent.getType(), instance.getServiceId());
         switch (pathChildrenCacheEvent.getType()) {
             case CHILD_ADDED:
+                ScheduledFuture<?> task = delayedTasks.remove(instance.getServiceId());
```

Comment:
I am not sure that ```childEvent``` is executed synchronously. So this may cause corner cases if two events for the same server arrive simultaneously.